File: src/main/java/dev/arbor/gtnn/data/GTNNCreativeModeTabs.java
package dev.arbor.gtnn.data;

import com.gregtechceu.gtceu.common.data.GTCreativeModeTabs;
import com.tterrag.registrate.util.entry.RegistryEntry;
import dev.arbor.gtnn.GTNN;
import net.minecraft.world.item.CreativeModeTab;

import static dev.arbor.gtnn.GTNNRegistries.REGISTRATE;

public class GTNNCreativeModeTabs {
    public static final RegistryEntry<CreativeModeTab> MAIN_TAB = REGISTRATE.defaultCreativeTab("main",
                    builder -> builder.displayItems(new GTCreativeModeTabs.RegistrateDisplayItemsGenerator("main", REGISTRATE))
                            .title(REGISTRATE.addLang("itemGroup", GTNN.id("main"), "GT--"))
                            .icon(GTNNMachines.CHEMICAL_PLANT::asStack)
                            .build())
            .register();

}

File: src/main/java/dev/arbor/gtnn/data/GTNNCreativeModeTabs.kt
package dev.arbor.gtnn.data

import com.gregtechceu.gtceu.common.data.GTCreativeModeTabs
import com.tterrag.registrate.util.entry.RegistryEntry
import dev.arbor.gtnn.GTNN.id
import dev.arbor.gtnn.GTNNRegistries.REGISTRATE
import net.minecraft.world.item.CreativeModeTab

object GTNNCreativeModeTabs {
    val MAIN_TAB: RegistryEntry<CreativeModeTab> = REGISTRATE.defaultCreativeTab(
        "main"
    ) { builder: CreativeModeTab.Builder ->
        builder.displayItems(GTCreativeModeTabs.RegistrateDisplayItemsGenerator("main", REGISTRATE))
            .title(REGISTRATE.addLang("itemGroup", id("main"), "GT--"))
            .icon { GTNNMachines.LargeNaquadahReactor.asStack() }
            .build()
    }.register()
}
